### PR TITLE
Add sfinv.get_homepage_name()

### PR DESCRIPTION
**Pull request by rubenwardy**
#1423

### CODE REVIEW

File: mods/sfinv/api.lua [hunk 1]
```diff
@@ -57,6 +56,10 @@ function sfinv.make_formspec(player, context, content, show_inv, size)
 	return table.concat(tmp, "")
 end
 
+function sfinv.get_homepage_name(player)
+	return sfinv.homepage_name or "sfinv:crafting"
```
Comment (rubenwardy) [inline mods/sfinv/api.lua]: this was done for backwards compatibility, although I'm now thinking it's not needed

**Comment by paramat** [comment]
Random thought for something simpler: How about automatically making the leftmost tab the homepage, then move creative 'all' tab to be leftmost? It seems intuitive that the homepage should be the leftmost tab.

File: mods/sfinv/api.lua (hunk 1 above)
Comment (paramat) [inline mods/sfinv/api.lua]: No need to support the sfinv which has been in use for a few days in a dev version.

Comment (paramat) [inline mods/sfinv/api.lua]: @rubenwardy could you remove this?

**Comment by paramat** [comment]
However i don't mind the homepage being specifiable.

**Comment by sfan5** [comment]
:+1: if the change requested by @paramat is done

**Comment by rubenwardy** [comment]
> Random thought for something simpler: How about automatically making the leftmost tab the homepage, then move the creative 'all' tab to be leftmost? It seems intuitive that the homepage should be the leftmost tab, i don't know why anyone would design a formspec any other way.

That makes it more complex. You'd need to add a way to allow mods to order pages (which would be rather clumsy), and then you'd still need a get_homepage to find the first page which is in the navigation.

For example, to make the tabs be like [all, nodes, items, tools, craft] you'd need to add a `sfinv.register_page_first()` and then insert the tabs in the _opposite_ order than what they will appear

**Comment by paramat** [comment]
Ok no problem, i don't understand the issues much. So i have no objection to this PR.

**Pull request merged**